pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -336,7 +337,7 @@ private RebalanceResult doRebalance(TableConfig tableConfig, RebalanceConfig reb
     //    current instances as this is the best we can do, and can help the table get out of this state.
     // 2. Only check the segments to be moved because we don't need to maintain available replicas for segments not
     //    being moved, including segments with all replicas OFFLINE (error segments during consumption).
-    Set<String> segmentsToMove = SegmentAssignmentUtils.getSegmentsToMove(currentAssignment, targetAssignment);
```

Comment:
Why change this from set to list?

Reply:
We only need to iterate through this list and there is no deduplication needed, so changed it to list for better performance